projects/plugins/jetpack/extensions/blocks/subscriptions/test/controls.js

```diff
@@ -0,0 +1,203 @@
+
+			expect( setButtonBackgroundColor.mock.calls[0][0] ).toMatch(/#[a-z0-9]{6,6}/);
+		} );
+	} );
+
+	describe( 'Color settings panel', () => {
+		test( 'hides gradient settings control panel', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } isGradientAvailable={ false }/> );
+
+			expect( screen.getByText( 'Background Colors' ) ).toBeInTheDocument();
+			expect( screen.queryByText( 'Color Settings' ) ).not.toBeInTheDocument();
+		} );
+
+		test( 'sets gradient background color', async () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+			userEvent.click( screen.getByText( 'Gradient' ) );
+			userEvent.click( screen.queryAllByLabelText( /Gradient\:/i, { selector: 'button' } )[0] );
+
+			expect( setGradient.mock.calls[0][0] ).toMatch(/linear\-gradient\((.+)\)/);
+		} );
+	} );
+
+	describe( 'Text settings panel', () => {
+		test( 'displays correctly', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+
+			expect( screen.getByText( 'Text Settings' ) ).toBeInTheDocument();
+		} );
+
+		test( 'set custom text ', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+			userEvent.click( screen.getByText( 'Text Settings' ), { selector: 'button' } );
+			userEvent.type( screen.getAllByLabelText( 'Custom Size' )[1], '18' );
+
+			expect( setAttributes ).toHaveBeenLastCalledWith( {
+				fontSize: 18,
+				customFontSize: 18,
+			} );
+		} );
+	} );
+
+	describe( 'Border settings panel', () => {
+		test( 'displays correctly', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+
+			expect( screen.getByText( 'Border Settings' ) ).toBeInTheDocument();
+		} );
+
+		test( 'set border radius', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+			userEvent.click( screen.getByText( 'Border Settings' ), { selector: 'button' } );
+			userEvent.type( screen.getAllByLabelText( 'Border Radius' )[1], '5' );
+
+			expect( setAttributes ).toHaveBeenLastCalledWith( {
+				borderRadius: 5,
+			} );
+		} );
+
+		test( 'set border weight', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+			userEvent.click( screen.getByText( 'Border Settings' ), { selector: 'button' } );
+			userEvent.type( screen.getAllByLabelText( 'Border Weight' )[1], '10' );
+
+			expect( setAttributes ).toHaveBeenLastCalledWith( {
+				borderWeight: 10,
+			} );
+		} );
+	} );
+
+	describe( 'Spacing settings panel', () => {
+		test( 'displays correctly', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+
+			expect( screen.getByText( 'Spacing Settings' ) ).toBeInTheDocument();
+		} );
+
+		test( 'set space inside', () => {
+			render( <SubscriptionsInspectorControls { ...defaultProps } /> );
+			userEvent.click( screen.getByText( 'Spacing Settings' ), { selector: 'button' } );
+			// For some reason userEvent.type won't work on these inputs
```

Comment:
Looks like this and the below one might not work with `userEvent.type` because the default value is double-digits and the max value for the input field is also double digits. If I change the the props we render the `SubscriptionsInspectorControls` with in the test to start with a `padding` of `0`, then I can get it to work:

```
			const propsWithPaddingOverride = { ...defaultProps, padding: 0 };
			render( <SubscriptionsInspectorControls { ...propsWithPaddingOverride } /> );
			userEvent.click( screen.getByText( 'Spacing Settings' ), { selector: 'button' } );
			userEvent.type( screen.getAllByLabelText( 'Space Inside' )[1], '13' );

			expect( setAttributes ).toHaveBeenLastCalledWith( {
				padding: 13,
			} );
```

I think it's fine to stick with `userEvent.clear` to test that it calls `setAttributes`, I just wanted to work out why it wasn't working 🙂

Reply:
Nice. Fresh 👀. Thanks!